fontforge/python.c

```diff
@@ -8521,49 +8540,76 @@ return(NULL);
 Py_RETURN( self );
 }
 
-/* PostScript importing flags */
+static char *glyph_import_keywords[] = { "filename", "correctdir",
+    "simplify", "handle_clip", "handle_eraser", "scale", "accuracy",
+    "default_joinlimit", "usesystem", "asksystem", NULL };
+
+/* Legacy PostScript importing flags */
 static struct flaglist import_ps_flags[] = {
     { "toobigwarn", 0 },			/* Obsolete */
     { "removeoverlap", 0 },			/* Obsolete */
-    { "handle_eraser", sf_handle_eraser },
-    { "correctdir",  sf_correctdir },
+    { "handle_eraser", 1 },
+    { "correctdir",  2 },
     FLAGLIST_EMPTY /* Sentinel */
 };
 
-static PyObject *PyFFGlyph_import(PyObject *self, PyObject *args) {
+static PyObject *PyFFGlyph_import(PyObject *self, PyObject *args,
+                                  PyObject *keywds) {
     SplineChar *sc = ((PyFF_Glyph *) self)->sc;
+    ImportParams ip, *ipp;
     char *filename;
     char *locfilename = NULL, *pt;
     PyObject *flags=NULL;
+    int psflags, use_system = false, ask_system = false;
+    bigreal jl_tmp = -1;
 
-    if ( !PyArg_ParseTuple(args,"es|O","UTF-8",&filename, &flags) )
-return( NULL );
+    InitImportParams(&ip);
+
+    if ( !PyArg_ParseTupleAndKeywords(args, keywds,
+                "es|$pppppddpp", glyph_import_keywords, "UTF-8", &filename,
```

Comment:
Can we avoid overloading this method? You more than likely wouldn't do this if you were writing this in pure python. How about:

```
es|O$pppppddpp
```

And check if flags is null or not?

Reply:
I don't think the interface should support simultaneous competing means of specifying the same flags and this is cleaner (and more comprehensive) than examining the keyword argument values to see if they've changed.